src/stream/src/executor/dml.rs

```diff
@@ -120,12 +149,62 @@ impl DmlExecutor {
                                 _ => {}
                             }
                         }
+
+                        // Stop barrier. It could be issued by scaling or DDLs(e.g. add column).
+                        if barrier.is_stop(self.actor_id) {
+                            self.dml_manager
+                                .unregister_changes_sender(self.table_id, self.actor_id);
+                        }
```

Comment:
This requires the executor to be gracefully stopped, which I guess cannot handle all cases. 😕

Personally I still suggest doing this clean-up stuff lazily when retreving a `WriteHandle`. We can keep the `loop` there but check whether the channel `is_closed` before returning it as the `WriteHandle`.

Reply:
If the executor is not gracefully stopped, does it means the recovery must happen? If it is true, I think we can just handle the graceful stop case. Anyway, I don't mind doing the clean-up stuff lazily, but I want to figure out what is the case that could result in an invalid `sender` without gracefully stopping.

Reply:
I'm not quite sure about that. Currently, there's no executor handling the `Stop` message explicitly except for the dispatcher and merger. Therefore, following this convention would be the safest approach. 😂